app/services/checksum_validator.rb

```diff
@@ -92,141 +87,202 @@ def moab_validator
+      @checksum_validator = checksum_validator
     end
-  end
 
-  # @param [Moab::VerificationResult] subentity
-  def parse_verification_subentity(subentity)
-    add_cv_result_for_modified_xml(subentity) if subentity.subsets[MODIFIED]
-    add_cv_result_for_additions_in_xml(subentity) if subentity.subsets[ADDED]
-    add_cv_result_for_deletions_in_xml(subentity) if subentity.subsets[DELETED]
-  end
+    # Adds to the AuditResults object for any errors in checksum validation it encounters.
+    def validate
+      return if manifest_inventory_verification_result.verified
 
-  def add_cv_result_for_modified_xml(subentity)
-    subentity.subsets.dig(MODIFIED, FILES).each_value do |details|
-      results.add_result(
-        AuditResults::MOAB_FILE_CHECKSUM_MISMATCH,
-        file_path: "#{subentity.details['other']}/#{details['basis_path']}",
-        version: subentity.details['basis']
-      )
+      manifest_inventory_verification_result.subentities.each do |subentity|
+        parse_verification_subentity(subentity) unless subentity.verified
+      end
+    rescue Nokogiri::XML::SyntaxError
+      results.add_result(AuditResults::INVALID_MANIFEST, manifest_file_path: manifest_file_path)
+    rescue Errno::ENOENT
+      results.add_result(AuditResults::MANIFEST_NOT_IN_MOAB, manifest_file_path: manifest_file_path)
     end
-  end
 
-  def add_cv_result_for_additions_in_xml(subentity)
-    subentity.subsets.dig(ADDED, FILES).each_value do |details|
-      results.add_result(
-        AuditResults::FILE_NOT_IN_MANIFEST,
-        file_path: "#{subentity.details['other']}/#{details['other_path']}",
-        manifest_file_path: "#{subentity.details['other']}/#{MANIFESTS_XML}"
-      )
+    private
+
+    attr_reader :moab_version, :checksum_validator
+
+    delegate :results, to: :checksum_validator
+
+    def manifest_file_path
+      @manifest_file_path ||= "#{moab_version.version_pathname}/#{MANIFESTS}/#{MANIFESTS_XML}"
     end
-  end
 
-  def add_cv_result_for_deletions_in_xml(subentity)
-    subentity.subsets.dig(DELETED, FILES).each_value do |details|
-      results.add_result(
-        AuditResults::FILE_NOT_IN_MOAB,
-        file_path: "#{subentity.details['other']}/#{details['basis_path']}",
-        manifest_file_path: "#{subentity.details['other']}/#{MANIFESTS_XML}"
-      )
+    def manifest_inventory_verification_result
+      @manifest_inventory_verification_result ||= moab_version.verify_manifest_inventory
     end
-  end
 
-  def validate_signature_catalog_entry(entry)
-    unless entry.signature.eql?(calculated_signature(signature_catalog_entry_path(entry)))
-      mismatch_error_data = { file_path: signature_catalog_entry_path(entry), version: entry.version_id }
-      results.add_result(AuditResults::MOAB_FILE_CHECKSUM_MISMATCH, mismatch_error_data)
+    # @param [Moab::VerificationResult] subentity
+    def parse_verification_subentity(subentity)
+      add_cv_result_for_modified_xml(subentity) if subentity.subsets[MODIFIED]
```

Comment:
and here the need for abbreviations definitely rears its head.

Reply:
Removing `_cv_`. It doesn't add anything here (and I think should be `AuditResult` anyway).